src/wp-includes/media.php

```diff
@@ -5177,3 +5191,67 @@ function wp_get_webp_info( $filename ) {
 
 	return compact( 'width', 'height', 'type' );
 }
+
+/**
+ * Gets the default value to use for a `loading` attribute on an element.
+ *
+ * This function should only be called for a tag and context if lazy-loading is generally enabled.
+ *
+ * The function usually returns 'lazy', but uses certain heuristics to guess whether the current element is likely to
+ * appear above the fold, in which case it returns a boolean `false`, which will lead to the `loading` attribute being
+ * omitted on the element. The purpose of this refinement is to avoid lazy-loading elements that are within the initial
+ * viewport, which can have a negative performance impact.
+ *
+ * Under the hood, the function uses a global `$wp_content_media_count` variable which it increases every time it is
+ * called for an element within the main content. If the element is the very first content element, the `loading`
+ * attribute will be omitted. This default threshold of 1 content element to omit the `loading` attribute for can be
+ * customized using the {@see 'wp_omit_loading_attr_threshold'} filter.
+ *
+ * @since 5.9.0
+ *
+ * @global int $wp_content_media_count Content media count used to enumerate elements for the contexts 'the_content'
+ *                                     and 'the_post_thumbnail'.
+ *
+ * @param string $context Context for the element for which the `loading` attribute value is requested.
+ * @return string|bool The default `loading` attribute value. Either 'lazy', 'eager', or a boolean `false`, to indicate
+ *                     that the `loading` attribute should be skipped.
+ */
+function wp_get_loading_attr_default( $context ) {
+	global $wp_content_media_count;
```

Comment:
Yeah, I guess it may be fine to actually remove the awareness of `$wp_content_media_count` from the tests entirely. They may be a bit harder to follow when reading the tests, but some extra doc comments could help with that. The tests would still already test whether the correct value for the `loading` attribute is returned so it would be sufficient.

I'll update it to use a static, then we can't influence it from the tests, but per the above I think that's okay.

Reply:
I've introduced a simple function `wp_increase_content_media_count()` to handle the static count. Using just a static won't work as it cannot be influenced from the outside at all. And while we don't want it to be typically modified from the outside, preventing it entirely is not a good idea as it also prevents us from being able to test it.

I think using a function is an elegant approach that allows us to control the value while not making it too easy to e.g. accidentally override/modify it.

Reply:
Yeah, having `wp_increase_content_media_count()` makes it better, but the count can still be easily manipulated by any theme/plugin. I'll have a better look at the tests to see if this can be avoided.

The output of the `wp_omit_loading_attr_threshold` filter would need to be made into a `static` too to avoid running that filter multiple times. Currently it seems it may run 4-5 tomes per page load.